korman/exporter/etlight.py

```diff
@@ -384,25 +386,25 @@ def _prep_for_lightmap(self, bo, toggle):
             uv_textures.active = uvtex
 
             # if the artist hid any UVs, they will not be baked to... fix this now
-            bpy.ops.object.mode_set(mode="EDIT")
-            bpy.ops.uv.reveal()
-            bpy.ops.object.mode_set(mode="OBJECT")
+            with self._set_mode("EDIT"):
+                bpy.ops.uv.reveal()
             self._associate_image_with_uvtex(uv_textures.active, im)
-            bpy.ops.object.mode_set(mode="EDIT")
 
-            # prep the uvtex for lightmapping
-            bpy.ops.mesh.select_all(action="SELECT")
-            bpy.ops.uv.average_islands_scale()
-            bpy.ops.uv.pack_islands()
+            # Meshes with modifiers need to have islands packed to prevent generated vertices
+            # from sharing UVs. Sigh.
+            if self.is_collapsed(bo):
```

Comment:
This never returns true. Looks like `LightBaker.__enter__` is never called due to lack of `with` statement. `MeshConverter.__enter__` is, though.

Reply:
OK, this is a bit of a chicken and egg problem because both `MeshConverter` and `LightBaker` are `_MeshManager`s. During the export process, this will always return False but it should work using the bake operators. I'll figure out what to do on that.